docs/DesignDocs/oe_log_message()_callback_proposal.md

```diff
@@ -0,0 +1,61 @@
+Proposal of Public Callback API for oe_log_message()
+====
+
+# Motivation
+
+This document describes the proposed public callback API for oe_log_message().
+
+There are several problems about the current logging that prevent end users to
+make full use of log:
+- It does't have mechanism to communicate between enclave and host.
+- Message can't be redirected to different destinations on demand.
+- Lack of method for end user to customize messages.
+
+The callback API is supposed to solve such problems.
+
+# User Experience
+
+By adding this public API, user can customize the output of the log message:
+- The output destination can be stdout/stderr or any specified files, and it is
```

Comment:
re: "files" plural, does it really support logging the same message to multiple files?

Reply:
Yes, the user can pass in multiple files if the context is a array of files, and the user have full control to write filtered messages to one specified file, or multiple files as he wish.